src/components/TagContainer.ts

```diff
@@ -0,0 +1,277 @@
+        const referenceTags = this.props.mxObject.getReferences(this.referenceAttribute) as string[];
+        const getSuggestions = tagObjects.map(object => ({ value: object.get(this.props.tagAttribute) as string }));
+
+        tagObjects.map(object => {
+            if (referenceTags.toString() !== "") {
+                referenceTags.map(reference => {
+                    if (reference === object.getGuid()) {
+                        currentTags.push(object);
+                    }
+                });
+            }
+        });
+        const getTags = currentTags !== []
+        ? currentTags.map(object => ({ value: object.get(this.props.tagAttribute) as string }))
+        : [];
+        this.setState({
+            fetchTags: true,
+            tagCache: tagObjects,
+            tagList: getTags.map(tag => tag.value),
+            suggestions: getSuggestions.map(suggestion => suggestion.value)
+        });
+    }
+
+    private createTag(newTag: string) {
+        const { afterCreateMicroflow, onChangeMicroflow, mxObject } = this.props;
+        const tagList = this.state.tagList;
+
+        for (const object of this.state.tagCache) {
+            const tagValue = object.get(this.props.tagAttribute) as string;
+            if (newTag === tagValue && newTag.trim() !== "") {
+                tagList.push(newTag);
+                mxObject.addReference(this.referenceAttribute, object.getGuid());
+                this.saveTagData(mxObject);
+                this.setState({ tagList });
+
+                return;
+            }
+        }
+        mx.data.create({
+            callback: object => {
+                object.set(this.props.tagAttribute, newTag);
+                mx.data.commit({
+                    callback: () => {
+                        mxObject.addReference(this.referenceAttribute, object.getGuid());
+                        this.saveTagData(mxObject);
+                        if (afterCreateMicroflow || onChangeMicroflow) {
+                            this.executeAction(mxObject, afterCreateMicroflow);
+                            this.executeAction(mxObject, onChangeMicroflow);
+                        }
+                    },
+                    error: error => window.mx.ui.error("Error occurred attempting to commit: " + error.message),
+                    mxobj: object
+                });
+            },
+            entity: this.tagEntity,
+            error: error => window.mx.ui.error(`Error creating tag object ${this.tagEntity}, ${error.message}`)
+        });
+
+        tagList.push(newTag);
+        this.setState({ tagList });
+    }
+
+    private removeTag(value: string) {
+        if (value.trim() !== "") {
+            const { onChangeMicroflow, mxObject } = this.props;
+            const tagIndex = this.state.tagList.indexOf(value);
+            const tagList = this.state.tagList.slice();
+
+            mx.data.get({
+                callback: (object) => {
```

Comment:
argument types is array... names suggest it is not! rename to `objects`
also no need use the parenthesis in arrow function with single argument `callback: objects => {`